terra_notebook_utils/drs.py

```diff
@@ -168,40 +175,36 @@ def resolve_drs_for_gs_storage(drs_url: str) -> Tuple[gs.Client, DRSInfo]:
     client = gs.get_client(info.credentials, project=project_id)
     return client, info
 
+def _resolve_target(filepath: str, info: DRSInfo) -> str:
+    if os.path.isdir(filepath):
+        filename = info.name or info.key.rsplit("/", 1)[-1]
+        filepath = os.path.join(os.path.abspath(filepath), filename)
+    return filepath
+
 def copy_to_local(drs_url: str,
                   filepath: str,
                   workspace_name: Optional[str]=WORKSPACE_NAME,
                   workspace_namespace: Optional[str]=WORKSPACE_GOOGLE_PROJECT):
     """Copy a DRS object to the local filesystem."""
     assert drs_url.startswith("drs://")
     enable_requester_pays(workspace_name, workspace_namespace)
-    client, info = resolve_drs_for_gs_storage(drs_url)
-    blob = client.bucket(info.bucket_name, user_project=workspace_namespace).blob(info.key)
-    if os.path.isdir(filepath):
-        filename = info.name or info.key.rsplit("/", 1)[-1]
-        filepath = os.path.join(os.path.abspath(filepath), filename)
     logger.info(f"Downloading {drs_url} to {filepath}")
-    try:
-        with open(filepath, "wb") as fh:
-            blob.download_to_file(fh)
-    except Exception:
-        os.remove(filepath)
-        raise
+    info = get_drs_info(drs_url)
+    get_drs_blob(info, workspace_namespace).download(_resolve_target(filepath, info))
 
 def head(drs_url: str,
          num_bytes: int = 1,
-         buffer: int = MULTIPART_THRESHOLD,
-         workspace_name: Optional[str] = WORKSPACE_NAME,
-         workspace_namespace: Optional[str] = WORKSPACE_GOOGLE_PROJECT):
+         buffer: Optional[int]=None,
+         workspace_name: Optional[str]=WORKSPACE_NAME,
+         workspace_namespace: Optional[str]=WORKSPACE_GOOGLE_PROJECT):
     """Head a DRS object by byte."""
     assert drs_url.startswith("drs://"), f'Not a DRS schema: {drs_url}'
     enable_requester_pays(workspace_name, workspace_namespace)
     try:
-        client, info = resolve_drs_for_gs_storage(drs_url)
-        blob = client.bucket(info.bucket_name, user_project=workspace_namespace).blob(info.key)
-        with gscio.Reader(blob, chunk_size=buffer) as handle:
-            the_bytes = handle.read(num_bytes)
-
+        blob = get_drs_blob(drs_url, workspace_namespace)
+        chunk_size = buffer or 2 * num_bytes
```

Comment:
Is the default buffer size always twice as big as the actual data it's fetching (`2 * num_bytes`)?  Would this still work if it just defaults to the size of the data (`num_bytes`)?